### PR TITLE
Improve the release pipeline

### PR DESCRIPTION
**Pull request by lbalmaceda**
### Description

This PR is reusing some old scripts I had around, which I refactored a bit to include the docs and user-agent update, the git tag generation.

Note that this is not going to push the changes nor do `npm publish` for us. The idea is that the diff gets a chance to be reviewed **locally** in case the changelog didn't pick the right items. 

### Usage: Prepare release
1. Create a milestone `v{x.y.z}` on Github.
2. Use the `CH: xxxxx` labels in your issues or PRs, and add them to the previously created milestone.
3. Run `npm release {x.y.z}` and wait for it to finish.
4. Open a new PR from the `prepare/v{x.y.z}` branch against `master`, wait for the CI to pass, and peer approval review.
5. Merge against `master` and wait for the CI to pass again.

#### Errors
If the generated changelog is not correct, or you used the wrong version name, you'll need to delete both the tag and the branch locally before re-running the command.
1. Run `git tag -d v{x.y.z}` to delete the local tag.
2. Run `git branch -D prepare/{x.y.z}` to delete the local branch.
3. Run `npm run release:clean` to remove the temporary files, associated with the failed release.

### Usage: Publish to npm
1. Run `npm run release:publish`. 
2. Input the OTP code if prompted.

### Testing
Manually
- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "scripts": {
     "ng": "ng",
     "start": "ng serve",
-    "prebuid": "npm run update-useragent",
+    "prebuild": "npm run update-useragent",
```
Comment (lbalmaceda) [inline package.json:7]: this one drove me nuts 😂

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+**Early Access Release**
+Do not use it in a Production environment.
+
+### Installation
+
+In order to install this package on your application, download the `auth0-auth0-angular-0.1.0.tgz` file from the Releases section on the Github repository and run the following command:
+
+```bash
+npm install /path/to/auth0-auth0-angular-0.1.0.tgz
+```
```
Comment (stevehobbsdev) [inline CHANGELOG.md]: What is this content and where has it come from?

It's just it has references to SPA SDK and the content doesn't look like a typical changelog

**Review by stevehobbsdev — APPROVED** [review]
All looks good, just had a question about the changelog. Otherwise, please merge.

File: CHANGELOG.md (hunk 2 above)
Comment (lbalmaceda) [reply CHANGELOG.md]: Good 👀 . I needed to have something there before for the changelog script to work, so I added the first release and some notes. I'll fix the link to point to this repo.

**Pull request merged**